content/src/content/jcr_root/apps/core/wcm/components/accordion/v1/accordion/clientlibs/site/js/accordion.js

```diff
@@ -574,7 +575,9 @@
      */
     function onDocumentReady() {
         dataLayerEnabled = document.body.hasAttribute("data-cmp-data-layer-enabled");
-        dataLayer = (dataLayerEnabled) ? window.adobeDataLayer = window.adobeDataLayer || [] : undefined;
+        dataLayerName = document.body.getAttribute("data-cmp-data-layer-name");
+        dataLayerName = dataLayerName || "adobeDataLayer";
+        dataLayer = dataLayerEnabled ? window[dataLayerName] = window[dataLayerName] || [] : undefined;
```

Comment:
this line is quite confusing, would it be better to break it in 2 lines?, also undefined is the value already set, would it be better to just be an if statement or if you want it to do it in one line, use the && operator even though some people might consider that confusing as well, or maybe if you want to keep it as a ternary use the null value

Reply:
thanks @cbogdan0707 , I agree this has become all a bit twisted and thanks for picking it up.  How about this.  Not shorter but clearer and no unnecessary undefined declaration:

If (dataLayerEnabled) {
	dataLayerName = document.body.getAttribute("data-cmp-data-layer-name") || "adobeDataLayer";
	dataLayer = window[dataLayerName];
}

Reply:
hi @cbogdan0707, sorted!